ecommerce/models.py

```diff
@@ -58,3 +59,50 @@ class BasketItem(TimestampedModel):
         Basket, on_delete=models.CASCADE, related_name="basket_item"
     )
     quantity = models.PositiveIntegerField()
+
+
+class Discount(TimestampedModel):
+    """Discount model"""
+
+    amount = models.DecimalField(
+        decimal_places=5,
+        max_digits=20,
+    )
+    automatic = models.BooleanField(default=False)
+    discount_type = models.CharField(choices=DISCOUNT_TYPES, max_length=30)
+    redemption_type = models.CharField(choices=REDEMPTION_TYPES, max_length=30)
+    max_redemptions = models.PositiveIntegerField(null=True, default=0)
+
+    def __str__(self):
+        return f"{self.amount} {self.discount_type} {self.redemption_type}"
+
+
+class UserDiscount(TimestampedModel):
+    """pre-assignment for a discount to a user"""
+
+    discount = models.ForeignKey(
+        Discount, on_delete=models.CASCADE, related_name="user_discount_discount"
+    )
+    user = models.ForeignKey(
+        User, on_delete=models.CASCADE, related_name="user_discount_user"
```

Comment:
As a [best practice](https://docs.djangoproject.com/en/4.0/topics/auth/customizing/#referencing-the-user-model),  `settings.AUTH_USER_MODEL` should be used here:
```suggestion
        settings.AUTH_USER_MODEL, on_delete=models.CASCADE, related_name="user_discount_user"
```

Reply:
Should be fixed in ad31885b13c9acb18693d2edf90d1778735ecdf9